python/jiminy_py/src/jiminy_py/viewer/viewer.py

```diff
@@ -1405,6 +1415,7 @@ def set_legend(labels: Optional[Sequence[str]] = None) -> None:
                         robot_name, color_text, text)
 
     @staticmethod
+    @_is_async
     @_must_be_open
     @_with_lock
```

Comment:
`_with_lock` must be top-most decorator everywhere, and `_is_async` the bottom-most.